mithril-client-cli/src/commands/cardano_db_v2/download.rs

```diff
@@ -216,19 +218,46 @@ impl CardanoDbV2DownloadCommand {
         Ok(())
     }
 
+    fn compute_required_disk_space_for_snapshot(
+        cardano_db: &CardanoDatabaseSnapshot,
+        restoration_options: &RestorationOptions,
+    ) -> u64 {
+        if restoration_options.immutable_file_range == ImmutableFileRange::Full {
+            cardano_db.total_db_size_uncompressed
+        } else {
+            let total_immutables_restored = restoration_options
+                .immutable_file_range
+                .length(cardano_db.beacon.immutable_file_number);
+            let total_immutables_restored_size =
+                total_immutables_restored * cardano_db.immutables.average_size_uncompressed;
+            let total_size = if restoration_options
```

Comment:
We may separate what is always on total_size and what is optional.
We can do it with a mutable total or using `then`, `unwrap`
```suggestion
            let mut total_size =
                total_immutables_restored_size + cardano_db.digests.size_uncompressed;
            if restoration_options
                .download_unpack_options
                .include_ancillary
            {
                total_size += cardano_db.ancillary.size_uncompressed;
            }
```

```suggestion
            let total_size = total_immutables_restored_size
                + cardano_db.digests.size_uncompressed
                + include_ancillary
                    .then(|| cardano_db.ancillary.size_uncompressed)
                    .unwrap_or(0);
```

Reply:
Updated by adding a dedicated function to compute the total immutables restored size.